AWScliSetup.sh

```diff
@@ -25,76 +25,125 @@ SYSTEMDSVCS=(
+   else
+      echo "Downloaded ${ZIPSRC} to ${AWSZIP}."
+   fi
 
-# Unzip the AWScli bundle into /tmp
-(cd /tmp && unzip -o ${AWSZIP})
+   # Unzip the AWScli bundle into /tmp
+   (cd /tmp && unzip -o ${AWSZIP})
 
-# Copy the de-archived zip to ${CHROOT}
-cp -r /tmp/awscli-bundle "${CHROOT}/root"
+   # Copy the de-archived zip to ${CHROOT}
+   cp -r /tmp/awscli-bundle "${CHROOT}/root"
 
-# Install AWScli bundle into ${CHROOT}
-chroot "${CHROOT}" /root/awscli-bundle/install -i /opt/aws/cli -b /usr/bin/aws
+   # Install AWScli bundle into ${CHROOT}
+   chroot "${CHROOT}" /root/awscli-bundle/install -i /opt/aws/cli -b /usr/bin/aws
 
-# Verify AWScli functionality within ${CHROOT}
-chroot "${CHROOT}" /usr/bin/aws --version
+   # Verify AWScli functionality within ${CHROOT}
+   chroot "${CHROOT}" /usr/bin/aws --version
 
-# Cleanup
-rm -rf "${CHROOT}/root/awscli-bundle"
+   # Cleanup
+   rm -rf "${CHROOT}/root/awscli-bundle"
+}
 
-# Depending on RPMs dependencies, this may fail if a repo is
-# missing (e.g. EPEL). Will also fail if no RPMs are present
-# in the search directory.
-{ STDERR=$(yum install -y "${EPELRELEASE}" 2>&1 1>&$out); } {out}>&1 || echo "$STDERR" | grep "Error: Nothing to do"
-{ STDERR=$(yum --installroot="${CHROOT}" install -y "${EPELRELEASE}" 2>&1 1>&$out); } {out}>&1 || echo "$STDERR" | grep "Error: Nothing to do"
-yum --installroot="${CHROOT}" install -y "${SCRIPTROOT}"/AWSpkgs/*.el7.*.rpm \
-   || exit $?
+enable_services()
+{
+   # Need to force systemd services to be enabled in resultant AMI
+   for SVC in "${SYSTEMDSVCS[@]}"
+   do
+      if [ -f "${CHROOT}/etc/systemd/system/$SVC" ]
+      then
+         printf "Attempting to enable %s in %s... " "${SVC}" "${CHROOT}"
+         # shellcheck disable=SC2015
+         chroot "${CHROOT}" /usr/bin/systemctl enable "${SVC}" && echo "Success" || \
+         ( echo "FAILED" ; exit 1 )
+      fi
+   done
+}
 
-# Need to force systemd services to be enabled in resultant AMI
-for SVC in "${SYSTEMDSVCS[@]}"
-do
-   printf "Attempting to enable %s in %s... " "${SVC}" "${CHROOT}"
-   chroot "${CHROOT}" /usr/bin/systemctl enable "${SVC}" && echo "Success" || \
-     ( echo "FAILED" ; exit 1 )
-done
+get_awstools_filenames()
+{
+   if [ "${AWSTOOLSRPM:-""}" = "" ]
+   then
+      ls "${SCRIPTROOT}"/AWSpkgs/*.el7.*.rpm
+   else
+      rpmfiles=""
+      for rpmfile in ${AWSTOOLSRPM}
+      do
+         rpmfiles="${rpmfiles} "$(ls "${SCRIPTROOT}/AWSpkgs/${rpmfile}"*.el7.*.rpm)
+      done
+      echo "${rpmfiles}"
+   fi
+}
+
+install_awstools()
+{
+   # Depending on RPMs dependencies, this may fail if a repo is
+   # missing (e.g. EPEL). Will also fail if no RPMs are present
+   # in the search directory.
+   # use yum info .. to check and show details for installed RPMs.
+   rpmfiles=$(get_awstools_filenames)
+   if [ "${rpmfiles}" = "" ]
+   then
+      echo "No Installation of addional Amazon RPMs"
+   else
+      yum info epel-release >/dev/null 2>&1 || yum install -y "${EPELRELEASE}" 
```

Comment:
> i think your alternative approach is equivalent, but we'll need some time to really exercise it

ahh, no. i just remembered the other caveat. we use this project in a completely disconnected environment, where our epel is an internal mirror. we maintain a similar release rpm that installs the repo definition, but the package name is not `epel-release` in order to avoid conflicts. we also support providing just a URL for `EPELRELEASE`. therefore, we cannot _know_ the package name in advance, and this change will not work (at least not without yet another variable for the epel repo name)...

Reply:
What about the following idea:
- download the epel-rpm
- extract name of RPM from download
- check installation with yum info

The ShellChecker in my testenvironment wasn't able to check the previous code and it looked really complex for te stupid behavior of yum. I think that the check with yum info is a much more usable way.